lapce-proxy/src/buffer.rs

```diff
@@ -173,21 +173,78 @@ pub fn read_path_to_string_lossy<P: AsRef<Path>>(
+                    "coffee" => "coffeescript",
+                    "c" | "h" => "c",
+                    "cpp" | "hpp" | "cxx" | "hxx" | "c++" | "h++" | "cc" | "hh" => {
+                        "cpp"
+                    }
+                    "cs" | "csx" => "csharp",
+                    "css" => "css",
+                    "diff" | "patch" => "diff",
+                    "dart" => "dart",
+                    "dockerfile" => "dockerfile",
+                    "ex" | "exs" => "elixir",
+                    "erl" | "hrl" => "erlang",
+                    "fs" | "fsi" | "fsx" | "fsscript" => "fsharp",
+                    "git-commit" | "git-rebase" => "git",
+                    "go" => "go",
+                    "groovy" | "gvy" | "gy" | "gsh" => "groovy",
+                    "hbs" => "handlebars",
+                    "htm" | "html" | "xhtml" => "html",
+                    "ini" => "ini",
+                    "java" | "class" => "java",
+                    "js" => "javascript",
+                    "jsx" => "javascriptreact",
+                    "json" => "json",
+                    "jl" => "julia",
+                    "less" => "less",
+                    "lua" => "lua",
+                    "makefile" | "gnumakefile" => "makefile",
+                    "md" | "markdown" => "markdown",
+                    "m" => "objective-c",
+                    "mm" => "objective-cpp",
+                    "plx" | "pl" | "pm" | "xs" | "t" | "pod" | "cgi" => "perl",
+                    "p6" | "pm6" | "pod6" | "t6" | "raku" | "rakumod"
+                    | "rakudoc" | "rakutest" => "perl6",
+                    "php" | "phtml" | "pht" | "phps" => "php",
+                    "ps1" | "ps1xml" | "psc1" | "psm1" | "psd1" | "pssc"
+                    | "psrc" => "powershell",
+                    "py" => "python",
+                    "r" => "r",
+                    "rb" => "ruby",
+                    "rs" => "rust",
+                    "scss" | "sass" => "scss",
+                    "sc" | "scala" => "scala",
+                    "sh" | "bash" | "zsh" => "shellscript",
+                    "sql" => "sql",
+                    "swift" => "swift",
+                    "toml" => "toml",
+                    "ts" => "typescript",
+                    "tsx" => "typescriptreact",
+                    "tex" => "tex",
+                    "vb" => "vb",
+                    "xml" => "xml",
+                    "xsl" => "xsl",
+                    "yml" | "yaml" => "yaml",
+                    _ => return None,
+                },
+            }
         }
-        "c" | "h" => "c",
-        "js" => "javascript",
-        "jsx" => "javascriptreact",
-        "ts" => "typescript",
-        "tsx" => "typescriptreact",
-        "ex" | "exs" => "elixir",
-        _ => return None,
+        // Handle paths without extension
+        None => match path.file_name()?.to_str()? {
+            "dockerfile" => "dockerfile",
```

Comment:
For Dockerfile's sometimes I like to have them with extensions for something like mutli-module jvm applications.
i.e: `Dockerfile.<module>` editor plugins like vscode's docker extensions support doing that so I think it would be nice if Lapce could as well. Though I feel like it should probably not clash with other existing language extensions. Probably a little niche but IMO a nice to have.

Reply:
This will be handled by detecting file content which lapce currently can't.

Reply:
> This will be handled by detecting file content which lapce currently can't.

Ah, alright :)